src/Sarif/UriHelper.cs

```diff
@@ -69,6 +71,16 @@ public static string MakeValidUri(string path)
                     !path.StartsWith(@"\", StringComparison.Ordinal))
                 {
                     validUri = validUri.Substring(1);
+
+                    // When the UriBuilder constructs an absolute URI, it strips any
+                    // leading "." and ".." segments ("dot-segments", as RFC 3986 calls
+                    // them). Glue them back on so we don't lose the relative path
+                    // information.
+                    string leadingDotSegments = GetLeadingDotSegments(path);
+                    if (!string.IsNullOrEmpty(leadingDotSegments))
+                    {
+                        validUri = leadingDotSegments + validUri;
+                    }
```

Comment:
The exposure in Roslyn to the non-absolute path is mostly theoretical since we don't expose the errorlogger except as a switch to the command-line compiler, which always provides absolute paths. Handling relative paths was a defense against future use of the error logger against arbitrary SourceText.Path values (which are possible if you do the analysis via the API and not the command line).

My question wasn't about this bug but why this code path for relative case was different than what I wrote even before this change. Did you fix another bug in porting my code here originally? What was it?

I think we should sync up the helpers to be exactly the same. My head hurts trying to understand what's different and if that's OK. I will also file a bug against corefx to get a way for System.Uri to just do this for us, because getting this right with worakarounds is proving to be quite complicated!